### PR TITLE
[master < T111-MAGE_memgraph-cugraph] cuGraph documentation

### PR DESCRIPTION
**Pull request by jmatak**

### CODE REVIEW

File: mage/query-modules/cuda/cugraph.md [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+### Procedures
+
+### `get(tolerance, max_iterations, normalized)`
+
+Find HITS authority and hub values for all nodes in the graph.
+The HITS algorithm computes two numbers for each node: its _authority_, which estimates the value of its content, and
+its _hub value_, which estimates the value of its links to other nodes.
+
+Whereas the HITS algorithm was designed for directed graphs, this implementation does not check if the input graph is
+directed and will execute on undirected graphs.
```
Comment (antoniofilipovic) [inline mage/query-modules/cuda/cugraph.md:126]: >  this implementation does not check if the input graph is directed and will execute on undirected graphs

But graphs in memgraph are always directed, how can user send an undirected graph, except if user creates one himself with both edge directions (a->b, b->a)

Comment (jmatak) [reply mage/query-modules/cuda/cugraph.md:126]: There is an argument '`directed`' that can be set to `true` or `false` while calling the algorithm

File: mage/installation/cugraph.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+If building MAGE × NVIDIA cuGraph locally, these requirements apply (tested on Ubuntu):
+
+- [**CMake**](https://cmake.org/) version above 3.20
+- [**NVIDIA CUDA developer toolkit**](https://developer.nvidia.com/cuda-toolkit) – CUDA version 11.6
+- System dependencies: `libblas-dev`, `liblapack-dev`, `libboost-all-dev`
+- [**NVIDIA NCCL communications library**](https://developer.nvidia.com/nccl)
+
+## Pulling the Docker image from DockerHub
+
```
Comment (antoniofilipovic) [inline mage/installation/cugraph.md:37]: I think something is missing here, like

To start `mage-cugraph` image, use the following command:

File: mage/installation/cugraph.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+```shell
+docker run -p 7444:7444 memgraph/memgraph-mage:1.3-cugraph-22.02-cuda-11.5
+```
+
+* Depending on your environment, different versions of MAGE/cuGraph/CUDA can be installed:
+```shell
+docker run -p 7444:7444 memgraph/memgraph-mage:${MAGE_VERSION}-cugraph-${CUGRAPH_VERSION}-cuda-${CUDA_VERSION}
+```
+
```
Comment (antoniofilipovic) [inline mage/installation/cugraph.md:46]: Add here maybe:

" To check which versions are available, jump to Docker Hub. There we have few options from which you can choose"

File: mage/installation/cugraph.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+:::info
+The development image with cuGraph support is not available yet. If you want to develop cuGraph-powered query
+modules in Docker, do not hesitate to [contact us](https://memgraph.com/community) about it.
+:::
+
+## Installing MAGE with NVIDIA cuGraph for Docker
```
Comment (antoniofilipovic) [inline mage/installation/cugraph.md:52]: "Building MAGExNvidia locally using Docker"
Maybe something like this.

File: mage/installation/cugraph.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    ```
+
+2.  Build the **MAGE × cuGraph**-tagged Docker image:
+    ```shell
+    docker build -f Dockerfile.cugraph -t memgraph-mage .
+    ```
+
+3.  Start Memgraph-MAGE with the following command:
+    ```shell
+    docker run --rm -p 7444:7444 --name mage memgraph-mage
```
Comment (antoniofilipovic) [inline mage/installation/cugraph.md:66]: `-p 7687:7687` again, and `--gpu all`?

File: mage/templates/_mage_spells.mdx [hunk 6]
```diff
@@ -1,21 +1,22 @@
...
 | [community_detection_online](/mage/query-modules/cpp/community-detection-online)       | C++    | A dynamic community detection algorithm suitable for large-scale graphs based upon label propagation. Runs in O(m) time and has O(mn) space complexity.                                                                           |
 | [cycles](/mage/query-modules/cpp/cycles)                                               | C++    | Algorithm for detecting cycles on graphs.                                                                                                                                                                                         |
+| [cugraph](/mage/query-modules/cpp/cugraph)                                               | CUDA   | Collection of NVIDIA GPU-powered algorithms integrated in Memgraph. Includes centrality measures, link analysis and graph clusterings.                                                                                            |
```
Comment (antoniofilipovic) [inline mage/templates/_mage_spells.mdx:11]: `cuda/cugraph`, not `cpp/cugraph`

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]

File: mage/installation/cugraph.md (hunk 2 above)
Comment (jmatak) [reply mage/installation/cugraph.md:37]: Also we were missing the pull command

File: mage/installation/cugraph.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+- [**CMake**](https://cmake.org/) version above 3.20
+- [**NVIDIA CUDA developer toolkit**](https://developer.nvidia.com/cuda-toolkit) – CUDA version 11.6
+- System dependencies: `libblas-dev`, `liblapack-dev`, `libboost-all-dev`
+- [**NVIDIA NCCL communications library**](https://developer.nvidia.com/nccl)
+
+## Pulling the Docker image from DockerHub
+
+```shell
+docker run -p 7444:7444 memgraph/memgraph-mage:1.3-cugraph-22.02-cuda-11.5
```
[the remark this answers is not in the document]
Comment (jmatak) [reply mage/installation/cugraph.md:39]: Correct!

File: mage/query-modules/cuda/cugraph.md [hunk 8]
```diff
@@ -0,0 +1,358 @@
...
+
+#### Input:
+
+- `tolerance: float(1e-5)` ➡ HITS approximation tolerance (custom values not supported by NVIDIA cuGraph)
+- `max_iterations: int(100)` ➡ Maximum number of iterations before returning an answer
+  (custom values not supported by NVIDIA cuGraph)
+- `normalized: bool(True)` ➡ Normalize the output (`False` not supported by NVIDIA cuGraph)
+
```
[the remark this answers is not in the document]
Comment (jmatak) [reply mage/query-modules/cuda/cugraph.md:160]: Making mistakes even while being concentrated

File: mage/installation/cugraph.md (hunk 2 above)
Comment (antepusic) [reply mage/installation/cugraph.md:37]: Good catch!

**Review by antepusic — APPROVED** [review]
Is it time to mark this as ready?

File: mage/installation/cugraph.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+If running MAGE × NVIDIA cuGraph in Docker, the following applies:
+
+- To run on NVIDIA-powered GPUs, RAPIDS requires Docker CE v19.03+ and
+  [**nvidia-container-toolkit**](https://github.com/NVIDIA/nvidia-docker#quickstart) installed.
+- Legacy Docker CE v17-18 users require the installation of the
+  [**nvidia-docker2**](https://github.com/NVIDIA/nvidia-docker/wiki/Installation-(version-2.0)) package.
+
+**Local build requirements:**
+
+If building MAGE × NVIDIA cuGraph locally, these requirements apply (tested on Ubuntu):
```
Comment (antepusic) [inline mage/installation/cugraph.md:29]: The Ubuntu version here is 20.04, right?

**Review by antoniofilipovic — APPROVED** [review]

**Review by g-despot — APPROVED** [review]

**Pull request merged**